### PR TITLE
Run Flutter FFI plugin template ffigen on CI

### PR DESCRIPTION
**Pull request by dcharkes**
```
  # Sanity check the latest `flutter create --template plugin_ffi`.
  # This will break if we change the Flutter template or the generated code.
  # But, getting libclang on the LUCI infrastructure has proven to be
  # non-trivial. See discussion on
  # https://github.com/flutter/flutter/issues/105513.
  # If we need to change the generated code, we should temporarily disable this
  # test, or temporarily disable the requirement for all bots to be green to
  # merge PRs.
  # Running this sanity check on one OS should be sufficient. Chosing Windows
  # because it is the most likely to break.
```

Integration test for:

* https://github.com/flutter/flutter/issues/105513

### CODE REVIEW

**Comment by dcharkes** [comment]
The test currently fails because

* https://github.com/dart-lang/ffigen/pull/397

is not in Flutter yet.

**Comment by dcharkes** [comment]
@liamappelbe PTAL

File: test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      print(message);
+      if (throwOnFailure) {
+        throw Exception(message);
+      }
+    }
+    print('Command `$commandString` done.');
+  }
+}
+
+class EnsureExists implements Task {
```
Comment (liamappelbe) [inline test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart:143]: This is unused

File: test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  test('Run Flutter', () async {
+    final projectDirUri = tempDirUri!.resolve('$projectName/');
+    final libDirUri = projectDirUri.resolve('lib/');
+    final bindingsGeneratedUri =
+        libDirUri.resolve('${projectName}_bindings_generated.dart');
+    final bindingsGeneratedCopyUri =
+        libDirUri.resolve('${projectName}_bindings_generated_copy.dart');
+
+    await Task.serial([
```
Comment (liamappelbe) [inline test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart:33]: All this Task stuff seems like a lot of boilerplate. Why not just write a `runProcess` and `copyFile` function, and do:

```dart
await runProcess(...);
await copyFile(...);
await runProcess(...);
```

**Review by liamappelbe — APPROVED** [review]
Approved with comments.

File: test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart (hunk 2 above)
Comment (dcharkes) [reply test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart:33]: I had this laying around in another project. I can simplify it for this test if you prefer.

Comment (liamappelbe) [reply test_flutter/flutter_template_tests/flutter_plugin_ffi_test.dart:33]: Feel free to merge as-is if you're too busy to clean it up. It's only a test after all. I've already approved.

**Pull request merged**